src/SimCore/Simulator.cxx

```diff
@@ -172,56 +186,44 @@ void Simulator::beforeNewRun(ldmx::RunHeader& header) {
                        "postInitCommands", {}));
 
-  auto bops{PluginFactory::getInstance().getBiasingOperators()};
-  for (const XsecBiasingOperator* bop : bops) {
+  auto bops{simcore::PluginFactory::getInstance().getBiasingOperators()};
+  for (const simcore::XsecBiasingOperator* bop : bops ) {
     bop->RecordConfig(header);
   }
 
-  auto dark_brem{
-      parameters_.getParameter<framework::config::Parameters>("dark_brem")};
-  if (dark_brem.getParameter<bool>("enable")) {
-    // the dark brem process is enabled, find it and then record its
-    // configuration
-    G4ProcessVector* electron_processes =
-        G4Electron::Electron()->GetProcessManager()->GetProcessList();
-    int n_electron_processes = electron_processes->size();
-    for (int i_process = 0; i_process < n_electron_processes; i_process++) {
-      G4VProcess* process = (*electron_processes)[i_process];
-      if (process->GetProcessName().contains(
-              darkbrem::G4eDarkBremsstrahlung::PROCESS_NAME)) {
-        // reset process to wrapped process if it is biased
-        if (dynamic_cast<G4BiasingProcessInterface*>(process))
-          process = dynamic_cast<G4BiasingProcessInterface*>(process)
-                        ->GetWrappedProcess();
-        // record the process configuration to the run header
-        dynamic_cast<darkbrem::G4eDarkBremsstrahlung*>(process)->RecordConfig(
-            header);
-        break;
-      }  // this process is the dark brem process
-    }    // loop through electron processes
-  }      // dark brem has been enabled
+  auto apMass{parameters_.getParameter<double>("APrimeMass")};
+  if (apMass > 0) {
+    header.setFloatParameter("A' Mass [MeV]", apMass);
+    header.setFloatParameter(
+        "Dark Brem Global Bias",
+        parameters_.getParameter<double>("darkbrem_globalxsecfactor"));
+    header.setStringParameter(
+        "Dark Brem Vertex Library Path",
+        parameters_.getParameter<std::string>("darkbrem_madgraphfilepath"));
+    header.setIntParameter("Dark Brem Interpretation Method",
+                           parameters_.getParameter<int>("darkbrem_method"));
+  }
 
   auto generators{
-      parameters_.getParameter<std::vector<framework::config::Parameters>>(
-          "generators")};
+      parameters_.getParameter<std::vector<framework::config::Parameters>>("generators")};
   int counter = 0;
   for (auto const& gen : generators) {
     std::string genID = "Gen " + std::to_string(++counter);
     auto className{gen.getParameter<std::string>("class_name")};
     header.setStringParameter(genID + " Class", className);
 
-    if (className.find("simcore::ParticleGun") != std::string::npos) {
+    if (className.find("ParticleGun") != std::string::npos) {
       header.setFloatParameter(genID + " Time [ns]",
                                gen.getParameter<double>("time"));
-      header.setFloatParameter(genID + " Energy [GeV]",
+      header.setFloatParameter(genID + " Energy [MeV]",
```

Comment:
Undo this reversion to a bug.

Side Note: move all run header config dumping to the generators themselves.